safe/gui/tools/shake_grid/shake_grid.py

```diff
@@ -821,6 +837,99 @@ def set_contour_properties(self, input_file):
 
         layer.commitChanges()
 
+    def mmi_point_sampling(self, place_layer, raster_mmi_path):
+        """Get MMI information from ShakeMap Tiff on place location.
+
+        Get the MMI value on each cities/places from the generated tiff files.
+
+        :param place_layer: Point layer generated from distance calculation
+        :type place_layer: QgsVectorLayer
+
+        :param raster_mmi_path: Path to converted mmi raster.
+        :type raster_mmi_path: str
+
+        :return: Modified place layer containing MMI values
+        :rtype: QgsVectorLayer
+        """
+        shake_raster = QgsRasterLayer(raster_mmi_path, 'shake_raster')
+        shake_provider = shake_raster.dataProvider()
+
+        place_fields = place_layer.dataProvider().fields()
```

Comment:
`place_fields = place_layer.fields()` No need for to fetch the dataprovider

Reply:
OK